resources/views/masjid_profile/edit.blade.php

```diff
@@ -21,4 +31,125 @@
+          </button>
+        </div>
+        <div class="modal-body">
+          <div class="img-container">
+              <div class="row justify-content-center text-center">
+                  <div class="col-md-8 justify-content-center text-center">
+                      <img id="image" src="" alt="{{ Setting::get('masjid_name') ?? 'buku masjid'}}">
+                  </div>
+                  <div class="col-md-4 justify-content-center text-center">
+                      <div class="preview"></div>
+                  </div>
+              </div>
+          </div>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-secondary" data-dismiss="modal">{{__('app.cancel')}}</button>
+          <button type="button" class="btn btn-primary" id="crop">{{__('app.crop')}}</button>
+        </div>
+      </div>
+    </div>
+</div>
 @endsection
+
+@section('styles')
+    {{ Html::style(url('css/plugins/cropper.css')) }}
+@endsection
+
+@push('scripts')
+    {{ Html::script(url('js/plugins/cropper.js')) }}
+    {{ Html::script(url('js/plugins/noty.js')) }}
+    <script>
+        var $modal = $('#modal-masjid');
+        var image = document.getElementById('image');
+        var cropper;
+
+        $(document).on("change", "#masjid_logo_image", function(e){
+            var files = e.target.files;
+            var done = function (url) {
+                image.src = url;
+                $modal.modal('show');
+            };
+            var reader;
+            var file;
+            var url;
+            if (files && files.length > 0) {
+                file = files[0];
+                if (URL) {
+                    done(URL.createObjectURL(file));
+                } else if (FileReader) {
+                    reader = new FileReader();
+                    reader.onload = function (e) {
+                        done(reader.result);
+                    };
+                    reader.readAsDataURL(file);
+                }
+            }
+        });
+        $modal.on('shown.bs.modal', function () {
+            cropper = new Cropper(image, {
+                aspectRatio: 1,
+                viewMode: 3,
+                preview: '.preview'
+            });
+        }).on('hidden.bs.modal', function () {
+            cropper.destroy();
+            cropper = null;
+        });
+        $("#crop").click(function(){
+            canvas = cropper.getCroppedCanvas({
+                width: 200,
+                height: 200,
+            });
+            canvas.toBlob(function(blob) {
+                url = URL.createObjectURL(blob);
+                var reader = new FileReader();
+                reader.readAsDataURL(blob); 
+                reader.onloadend = function() {
+                    var base64data = reader.result;
+                    $.ajax({
+                        type: "POST",
+                        dataType: "json",
+                        url: "{{ route('ajax.masjid_profile.image')}}",
```

Comment:
URL nya kita pakai `api` saja, nanti di api.php nya di-update.

```suggestion
                        url: "{{ route('api.masjid_profile.image') }}",
```